From: Modestas Vainius <modax@debian.org>
Subject: Fix FTBFS on GNU/kFreeBSD (and likely Gnu/Hurd)
Forwarded: yes
Origin: vendor

The original failure [1] was caused by the fact that cxxabi.h execinfo.h etc.
includes where guarded more restrictively (e.g. additionally by  __linux) than
the code which needs them (just by e.g.
KWSYS_SYSTEMINFORMATION_HAS_CPP_DEMANGLE).  kFreeGNU has some of these headers
hence extend __linux ifdef to kFreeBSD and GNU Hurd as well.

Ideally, the code should be fixed to have the same level of ifdef protection in
all places.

[1] https://buildd.debian.org/status/fetch.php?pkg=cmake&arch=kfreebsd-amd64&ver=2.8.12-1&stamp=1383507045

---
 Source/kwsys/SystemInformation.cxx |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/Source/kwsys/SystemInformation.cxx
+++ b/Source/kwsys/SystemInformation.cxx
@@ -123,7 +123,7 @@ typedef int siginfo_t;
 # endif
 #endif
 
-#if defined(__linux) || defined (__sun) || defined(_SCO_DS)
+#if defined(__linux) || defined (__sun) || defined(_SCO_DS) || defined(__GLIBC__) || defined(__GNU__)
 # include <fenv.h>
 # include <sys/socket.h>
 # include <netdb.h>
